src/components/widgets/PAutoComplete.js

```diff
@@ -1,5 +1,42 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import { PInput, PListItem, PList } from '../elements'
-import React from 'react'
+import React, { useCallback } from 'react'
+
+const MemoizeItem = React.memo(({ data, handlePress, keyword }) => {
+  const highlightText = data.slice(0, keyword.length);
+  const nonHighlighText = data.slice(keyword.length)
+  return (
+    <PListItem
+      onPress={handlePress}
+      dataCallback={data}
+    >
+      <span className="font-weight-bold">{highlightText}</span>{nonHighlighText}
+    </PListItem>
+  )
+})
+
+const MemoizeList = React.memo(({keyword, onSelect, suggestions }) => {
+
+  const handlePress = useCallback((e, dataCallback) => {
+    onSelect && onSelect(e, dataCallback)
+  }, [])
+
+  return (
+    <PList>
+
+      {/* filter one more time in FE, in case the result from the previous AJAX still exist in the UI and the user typing too fast*/}
+      {/* so it will give better UX for the user as they see the list change accordingly immediately*/}
+      {/* when the AJAX return, it will update the latest search result */}
+      {/* maybe consider to use useMemo if the filter is too slow in future */}
+      {suggestions.filter((item) => {
+        return item.toLocaleLowerCase().startsWith(keyword.toLocaleLowerCase())
+      }).map((item) => (
```

Comment:
The idea is really good. But consider the cost to keep maintain the filter logic in 3 different places (BE side, caching layer and here), I don't think this is a long term solution.